### PR TITLE
Enabled HIDAPI support

### PR DESCRIPTION
**Pull request by josch1710**
I enabled HIDAPI support, which gives us game controllers via USB.
Also I cleaned up most compiler warnings.

### CODE REVIEW

**Comment by SilvanScherrer** [comment]
I need to break the first commit into parts. As doing one commit for several things is not our philosophie.

**Comment by josch1710** [comment]
Go for it.

**Comment by SilvanScherrer** [comment]
I Think i added them all. Please check

File: src/audio/os2/SDL_os2audio.c [hunk 1]
```diff
@@ -62,7 +62,7 @@ static ULONG _getEnvULong(const char *name, ULONG ulMax, ULONG ulDefault)
 static int _MCIError(const char *func, ULONG ulResult)
 {
     CHAR    acBuf[128];
-    mciGetErrorString(ulResult, acBuf, sizeof(acBuf));
+    mciGetErrorString(ulResult, (PSZ)acBuf, sizeof(acBuf));
```
Comment (sezero) [inline src/audio/os2/SDL_os2audio.c:65]: What warnings are you having without all these? Are they fixed by using `-DOS2EMX_PLAIN_CHAR` instead?

Comment (SilvanScherrer) [reply src/audio/os2/SDL_os2audio.c:65]: Yes you are right. I will use the original Version again snd add the define. 
Now i know why i should not just add a pull request w/o checking further

Comment (sezero) [reply src/audio/os2/SDL_os2audio.c:65]: > Yes you are right. I will use the original Version again and add the define.

OK - See the mainstream configure.ac and cmake script.

**Comment by SilvanScherrer** [comment]
all should be applied. some part where done different.

**Pull request closed without merging**